waspc/packages/deploy/src/providers/fly/setup/setup.ts

```diff
@@ -0,0 +1,70 @@
+import { $, cd } from 'zx'
+import crypto from 'crypto'
+import * as tomlHelpers from '../helpers/tomlFileHelpers.js'
+import { DeploymentInfo, IDeploymentInfo } from '../DeploymentInfo.js'
+import { IGlobalOptions } from '../IGlobalOptions.js'
+import { buildDirExists, cdToClientBuildDir, cdToServerBuildDir, lazyInit, waspSays } from '../helpers/helpers.js'
+
+export async function setup(baseName: string, region: string, options: IGlobalOptions) {
+  waspSays(`Setting up your Wasp app with Fly.io!`)
+
+  const buildWaspIfMissing = lazyInit(async () => {
+    if (!buildDirExists(options.waspDir)) {
+      cd(options.waspDir)
+      await $`wasp build`
+    }
+  })
+
+  const tomlFiles = tomlHelpers.getTomlFileInfo(options)
+  const deploymentInfo = new DeploymentInfo(baseName, region, options, tomlFiles)
+
+  if (tomlHelpers.serverTomlExistsInProject(tomlFiles)) {
+    waspSays(`${tomlFiles.serverTomlPath} exists. Skipping server setup.`)
+  } else {
+    await buildWaspIfMissing()
+    await setupServer(deploymentInfo)
+  }
+
+  if (tomlHelpers.clientTomlExistsInProject(tomlFiles)) {
+    waspSays(`${tomlFiles.clientTomlPath} exists. Skipping client setup.`)
+  } else {
+    await buildWaspIfMissing()
+    await setupClient(deploymentInfo)
+  }
+
+  waspSays(`Don't forget to create your database by running the "create-db" command.`)
+}
+
+async function setupServer(deploymentInfo: IDeploymentInfo) {
+  waspSays(`Setting up server app with name ${deploymentInfo.serverName()}`)
+
+  cdToServerBuildDir(deploymentInfo.options.waspDir)
+  tomlHelpers.deleteLocalToml()
+
+  // This creates the fly.toml file, but does not attempt to deploy.
+  await $`flyctl launch --no-deploy --name ${deploymentInfo.serverName()} --region ${deploymentInfo.region}`
+
+  tomlHelpers.copyLocalServerTomlToProject(deploymentInfo.tomlFiles)
+
+  const randomString = crypto.randomBytes(32).toString('hex')
+  await $`flyctl secrets set JWT_SECRET=${randomString} PORT=8080 WASP_WEB_CLIENT_URL=${deploymentInfo.clientUrl()}`
+
+  waspSays(`Server setup complete!`)
```

Comment:
Ensure this is on its own line. Ref: https://www.notion.so/wasp-lang/Fly-Deploy-comments-61de123caf2348aabb2ccd6d972c84ff